Source/controls/plrctrls.cpp

```diff
@@ -510,20 +510,23 @@ void Interact()
 	}
 
 	const Player &myPlayer = *MyPlayer;
+	Direction pdir = myPlayer._pdir;
+	const AxisDirection moveDir = GetMoveDirection();
+	const bool motion = moveDir.x != AxisDirectionX_NONE || moveDir.y != AxisDirectionY_NONE;
 
-	if (leveltype != DTYPE_TOWN && IsStandingGround()) {

```

Comment:
This part is confusing to me. It seems you consolidated the conditions to reduce nesting, but now you're computing all these additional variables that are completely irrelevant if the player is in town or not standing ground. I don't usually advocate for more nesting, but in this case it clarifies the scope of the computations and corresponding variables.

If you want to reduce nesting, then I recommend refactoring this part of the logic into its own function.

Reply:
Calculation of `bNear` requires `position`. Calculation of `position` requires `pdir` and `motion`. Calculation of `pdir` requires `motion`. Calculation of `motion` requires `moveDir`. It's a trainwreck.